var/spack/repos/builtin/packages/fms/package.py

```diff
@@ -66,6 +66,8 @@ class Fms(CMakePackage):
     )
 
     variant("shared", description="Build shared libraries", when="@2024.02:", default=False)
+    # To build a shared/dynamic library, both `pic` and `shared` are required:
+    requires("+pic", when="+shared", msg="The +shared variant requires +pic")
```

Comment:
Would it be possible to say that the variant pic is conditional on `~shared`?

Reply:
I may have misunderstood your question. Isn't `+pic` valid when `shared` is enabled or disabled? It is `+shared` that is requires `+pic`.

Reply:
Sure, but if you make:
```python
variant("pic", default=True, ..., when="~shared")
```
then you won't have a `+shared~pic` combination to start with - since the `pic` variant exists only when `~shared`.

Reply:
Will that also prevent `+shared+pic`? That's the combination we want.

Do you prefer something like this?
```
variant("shared", description="Build shared libraries", when="@2024.02: +pic", default=False)
```

Reply:
`variant("pic", when="~shared")` means the pic variant will only exist when shared is disabled. In that case, when `+shared` is enabled, _the pic variant doesn't exist, the PIC setting won't be touched by Spack, and therefore it will be on by default through CMake_ (which is consistent with what we want), and if `~shared` is set then, then pic can be configured as on or off (also what we want). Personally I have no preference between `requires` vs. `conflicts` vs. `variant(when=...)`.

Reply:
Isn't FMS CMake's `FPIC` OFF by default?
https://github.com/NOAA-GFDL/FMS/blob/6b6695b3f017807c3870c25a0fc6661342678d5f/CMakeLists.txt#L58-L59